charmcraft/main.py

```diff
@@ -358,6 +359,7 @@ def main(argv=None):
     # process
     try:
         env.ensure_charmcraft_environment_is_supported()
+        setup_parts()
```

Comment:
Is this setup expensive? Maybe we should setup parts only in the commands that use it? (for example, it doesn't make sense to setup parts if you do `charmcraft help` or `charmcraft version`).

Reply:
The setup is very lightweight and was designed to be called at the application start with practically no overhead.